docs/source/howto/plugin_codes.rst

```diff
@@ -178,40 +194,51 @@ If a particular parser should always be used by default for a given |CalcJob|, i
-These exit codes can be defined through the |spec| of the |CalcJob| that is used for that code, just as the inputs and output are defined.
-For example, the :py:class:`~aiida.calculations.arithmetic.add.ArithmeticAddCalculation` introduced in :ref:`"Interfacing external codes"<how-to:plugin-codes:interfacing>`, defines the following exit codes:
+So far, we have not spent much attention on dealing with potential errors that can arise when running external codes.
+However, there are lots of ways in which codes can fail to execute nominally.
+A |Parser| can play an important role in detecting such errors and reporting them to AiiDA, where :ref:`workflows <how-to:workflows>` can then decide how to proceed, e.g. by modifying input parameters and resubmitting the calculation.
+
+Parsers communicate errors through :ref:`exit codes<topics:processes:concepts:exit_codes>`, which are defined in the |spec| of the |CalcJob| they parse.
+Our :py:class:`~aiida.calculations.arithmetic.add.ArithmeticAddCalculation` example, defines the following exit codes:
 
 .. literalinclude:: ../../../aiida/calculations/arithmetic/add.py
     :language: python
     :start-after: start exit codes
     :end-before: end exit codes
     :dedent: 8
 
-Each ``exit_code`` defines an exit status (a positive integer), a label that can be used to reference the code in the |parse| method (through the ``self.exit_codes`` property, as seen below), and a message that provides a more detailed description of the problem.
-To use these in the |parse| method, you just need to return the corresponding exit code which instructs the engine to store it on the node of the calculation that is being parsed.
-The snippet of the previous section on :ref:`parsing the outputs<how-to:plugin-codes:parsing>` already showed two problems that are detected and are communicated by returning the corresponding the exit code:
+Each ``exit_code`` defines
+
+ * an exit status (a positive integer),
+ * a label that can be used to reference the code in the |parse| method (through the ``self.exit_codes`` property, as shown below), and
+ * a message that provides a more detailed description of the problem.
+
+In order to inform AiiDA about a failed calculation, simply return from the ``parse`` method the exit code that corresponds to the detected issue.
+Here is a more complete version of our |Parser|:
 
 .. literalinclude:: ../../../aiida/parsers/plugins/arithmetic/add.py
     :language: python
-    :lines: 27-33
-    :dedent: 8
+    :pyobject: ArithmeticAddParser
+
+It checks:
+
+ 1. Whether a retrieved folder is present.
+ 2. Whether the output file can be read (whether ``open()`` or ``read()`` will throw an ``OSError``).
+ 3. Whether the output file contains an integer.
+ 4. Whether the sum is negative.
+
+AiiDA stored the exit code returned by the |parse| method on the calculation node that is being parsed, from where it can then be inspected further down the line.
```

Comment:
```suggestion
AiiDA stores the exit code returned by the |parse| method on the calculation node that is being parsed, from where it can then be inspected further down the line.
```